src/Server/Resource/DoctrineResource.php

```diff
@@ -330,7 +344,10 @@ public function create($data)
             return $data;
         }
 
-        $entity = new $entityClass;
+        $entity = $this->entityFactory
+            ? $this->entityFactory->instantiate($entityClass)
```

Comment:
Does Instantiator interface really work here, since construction could depend on $data? Wouldn't also need to pass $data to factory to make this useful?

Reply:
I know what you mean, but this PR is at least a simple, backwards-compatible first step. In fact I thought this PR was merged already and I'm surprised it still isn't...